src/aquarium-optimized/Aquarium.h

```diff
@@ -35,10 +36,10 @@ class Model;
 
 enum BACKENDTYPE : short {
   BACKENDTYPEANGLE,
-  BACKENDTYPEDAWND3D12,
-  BACKENDTYPEDAWNMETAL,
-  BACKENDTYPEDAWNVULKAN,
+  BACKENDTYPEDAWN,
   BACKENDTYPED3D12,
+  BACKENDTYPEMETAL,
+  BACKENDTYPEVULKAN,
   BACKENDTYPEOPENGL,
   BACKENDTYPELAST
 };
```

Comment:
I hope to define it as below, which will simply the data strcuture and code. 
enum BACKENDTYPE {
    BACKENDTYPENONE = 0,
    BACKENDTYPEANGLE = 1 << 0,
    BACKENDTYPEDAWN = 1 << 1,
    BACKENDTYPEOPENGL = 1 << 2,
    BACKENDTYPED3D11 = 1 << 3,
    BACKENDTYPED3D12 = 1 << 4,
    BACKENDTYPEMETAL = 1 << 5,
    BACKENDTYPEVULKAN = 1 << 6,
    BACKENDTYPELAST = BACKENDTYPEVULKAN,
}

Reply:
And this is a third proposal for discussion:

```c++
enum BACKENDTYPE {
  // Use the lower n bits to represent 2^n backends, instead of 1<<n
  // bits for that many backends (imagine we would add more backends
  // like D3D9 in the future). They are numbered continuously based on
  // the fact of being mutual exclusive.
  BACKENDTYPEOPENGL = 0x0000,
  BACKENDTYPED3D11 = 0x0001,
  BACKENDTYPED3D12 = 0x0002,
  BACKENDTYPEMETAL = 0x0003,
  BACKENDTYPEVULKAN = 0x0004,

  // Set the high bits for ANGLE or DAWN
  BACKENDTYPEANGLE = 0x0100,
  BACKENDTYPEDAWN = 0x0200,
};
```

Reply:
Comparing with the original PR, another subtle side effect of the above proposals is that switch-case will be turned into a if-else chain, at least in `ContextFactory::createContext()` I think.

Reply:
I love this third proposal personally. Per my feeling, DAWN/ANGLE/NATIVE is not in the same set as D3D11/D3D12/D3D9 etc.
Using high low bit is helpful to store all info in a single value and with the masks we can easily extract the useful info.
And it can eliminate the weired combinations in [1];